File: utility/templates/solution_template.py
from typing import List
import sys


def read_input() -> List[str]:
    with open('./input', 'r') as file:
        return file.readlines()


def part_one():
    raise NotImplementedError


def part_two():
    raise NotImplementedError


def main() -> None:
    arg = sys.argv[1] if len(sys.argv) > 1 else "both"
    data = read_input()
    if arg == '1':
        print(part_one(data))
    elif arg == '2':
        print(part_two(data))
    else:
        print(part_one(data))
        print(part_two(data))


if __name__ == '__main__':
    main()

File: utility/templates/solution_template.ts
// Run with: ts-node solution.ts [1|2]  (or compile with tsc and run with node)
import * as fs from "fs";

function readInput(): string[] {
  const data = fs.readFileSync("./input", "utf8");
  // Keep empty lines if you prefer: remove the filter(Boolean) below
  return data.split(/\r?\n/);
}

function partOne(lines: string[]): string {
  // TODO: implement
  return `part 1 not implemented (${lines.length} lines)`;
}

function partTwo(lines: string[]): string {
  // TODO: implement
  return `part 2 not implemented (${lines.length} lines)`;
}

function main() {
  const lines = readInput();
  const arg = process.argv[2];
  if (arg === "1") {
    console.log(partOne(lines));
  } else if (arg === "2") {
    console.log(partTwo(lines));
  } else {
    console.log(partOne(lines));
    console.log(partTwo(lines));
  }
}

main();
